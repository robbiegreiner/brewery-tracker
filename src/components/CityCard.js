import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CityCard = ({ brewery, getBrewery }) => {
  if (brewery.brewery){
    return (
      <Link to={'/brewery/' + brewery.brewery.id}>
        <div className='beer-card' onClick={() => getBrewery(brewery.breweryId)}>
          <h2>{brewery.brewery.name}</h2>
          <h3>{brewery.locationTypeDisplay}</h3>
          {/* <img className='brewery-icon' src={brewery.images.large}></img> */}
          <h4>{brewery.streetAddress}</h4>
          <h4>{brewery.locality} , {brewery.region} {brewery.postalCode}</h4>
        </div>
      </Link>
    );
  } else {
    return (
      <h1>No Match</h1>
    );
  }
};

CityCard.propTypes = {
  brewery: PropTypes.object,
  getBrewery: PropTypes.func
};

export default CityCard;
