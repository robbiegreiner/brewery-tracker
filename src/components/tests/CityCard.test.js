import React from "react";
import CityCard from '../CityCard';
import { shallow } from "enzyme";

describe('CityCard component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<CityCard
    user={{}}
    writeUserData={mockFn}
    getBrewery={mockFn}
    brewery={{}}
    key={'vxySde'}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
