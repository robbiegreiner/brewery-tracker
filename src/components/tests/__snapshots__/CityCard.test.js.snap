// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityCard component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CityCard
    brewery={Object {}}
    getBrewery={[Function]}
    user={Object {}}
    writeUserData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "No Match",
    },
    "ref": null,
    "rendered": "No Match",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "No Match",
      },
      "ref": null,
      "rendered": "No Match",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
